packages/stories/src/gitgraph-js/6-custom-renders.stories.tsx

```diff
@@ -300,4 +306,235 @@ storiesOf("gitgraph-js/6. Custom renders", module)
+            // => they are set later, on tag update.
+            // We would need to make tag update happen before to solve it.
+            const offset = parseFloat(tag.getAttribute("data-offset") || "0");
+            const tagWidth = tag.getBBox().width + 2 * TAG_PADDING_X + offset;
+            x += tagWidth + padding;
+
+            // ** changed ** - keeping track of highest tag
+            highestTag = Math.max(highestTag, tag.getBBox().height)
+          });
+
+          if (message) {
+            // ** changed ** - moving the message down instead of to the right
+            this.moveElementByXY(message, this.commitMessagesX, highestTag+padding+8);
+            // this.moveElement(message, x);
+          }
+        });
+      }
+    }
+
+    return (
+        <GraphContainer>
+          {(graphContainer) => {
+
+            const gitgraph = createGitgraph(graphContainer, {
+                  generateCommitHash: createFixedHashGenerator(),
+                },
+                () => {
+                  // keep reference to renderer so we can call re-render whenever `selectedCommit` changes
+                  return new MessageBelowTagsRenderer(graphContainer);
+                });
+
+            var master = gitgraph.branch('master');
+            // Tag on branch
+            master
+              .commit()
+              .tag('v1.0')
+              .tag('first release');
+            master.commit();
+            master.tag('v1.1');
+            master.commit({ tag: 'v1.2' });
+            // Tag on gitgraph
+            master.commit("This is good for narrower diagrams");
+            gitgraph.tag('v2.0');
+            // Custom tags
+            var customTagStyle = {
+              bgColor: 'orange',
+              strokeColor: 'orange',
+              borderRadius: 0,
+              pointerWidth: 0,
+            };
+            gitgraph.tag({
+              name: 'last release',
+              style: customTagStyle,
+            });
+            gitgraph
+              .branch('feat1')
+              .commit("Message appears below the tags")
+              .tag({ name: 'something cool', style: customTagStyle });
+          }}
+        </GraphContainer>
+    )
+  })
+  .add("custom renderer - background rect", () => {
+    // Providing a custom Renderer class can enable limitless customization.
+    // However -
+    // NOTE: using a custom Renderer can be considered advanced and experimental.
+    //   1. You will have to read the code to figure out which methods to override.
+    //   2. The internal API (ie. the Renderer methods and variables) is *internal* and
+    //      may change in the future.
+    //
+    // In this example we add a background rectangle to the commit which will change
+    // fill-color on hover and if selected.
+    //
+    // HOVER over the commits, and CLICK to see the background change
+
+    let selectedCommit: Commit|null = null;
+    let renderer: any;
```

Comment:
I think the proper type should be:

```suggestion
    let renderer: Renderer | undefined;
```

Reply:
😳